src/app/pages/TokenDashboardPage/TokenSupplyCard.tsx

```diff
@@ -0,0 +1,43 @@
+import { FC } from 'react'
+import { useTranslation } from 'react-i18next'
+import Box from '@mui/material/Box'
+import Typography from '@mui/material/Typography'
+import { SnapshotCard } from '../../components/Snapshots/SnapshotCard'
+import { COLORS } from '../../../styles/theme/colors'
+import { useRequiredScopeParam } from '../../hooks/useScopeParam'
+import { useTokenInfo } from './hook'
+import { useLoaderData } from 'react-router-dom'
+
+export const TokenSupplyCard: FC = () => {
+  const { t } = useTranslation()
+  const scope = useRequiredScopeParam()
+
+  const address = useLoaderData() as string
+
+  const { token, isFetched } = useTokenInfo(scope, address)
+
+  const title = t('tokens.totalSupply')
+  const supplyString = token?.total_supply || '0'
+  const supplyNumber = parseInt(supplyString)
```

Comment:
This could in theory throw an error. Would be nice to wrap it into try, catch

Reply:
For invalid data, this will return NaN. Now I have added different output depending on a check with `isNaN`.